// The file defines protos used internally with CLgen components.

syntax = "proto2";
package clgen;

import "deeplearning/clgen/proto/corpus.proto";
import "deeplearning/clgen/proto/model.proto";
import "deeplearning/clgen/proto/sampler.proto";

// The schema for a corpus metafile. An instance of this proto is created with
// name META.pbtxt in the cache directory of each corpus.
message CorpusMeta {
  optional Corpus config = 1;
  optional int32 preprocess_time_ms = 2;
}

// The schema for a model metafile. An instance of this proto is created with
// name META.pbtxt in the cache directory of each model.
message ModelMeta {
  optional Model config = 1;
  repeated ModelTrainingEpochStats training_stats = 2;
}

message ModelTrainingEpochStats {
  optional int32 batch_num = 1;
  optional int32 time_ms = 2;
  optional double training_cost = 3;
}

// The schema for a sampler metafile. An instance of this proto is created with
// name META.pbtxt in the cache directory of each model that samples it.
// TODO(cec): Re-implement sampler metafile creation and comparison.
message SamplerMeta {
  optional Sampler config = 1;
}

// A preprocessor worker input.
message PreprocessorWorkerJob {
  optional string contentfile_id = 1;
  optional string src = 2;
  repeated string preprocessors = 3;
}

// The output of a preprocessor worker.
message PreprocessorWorkerJobOutcome {
  enum Status {
    OK = 0;
    FAIL = 1;
  }

  optional string contentfile_id = 1;
  optional string contents = 2;
  optional Status status = 3;
}
